sql/src/test/java/org/apache/druid/sql/calcite/http/SqlResourceTest.java

```diff
@@ -689,6 +689,18 @@ private void checkSqlRequestLog(boolean success)
     }
   }
 
+  @Test
+  public void testExtractMillisecondFromExpr() throws Exception
```

Comment:
I think `CalciteQueryTests` might be a more appropriate place to test this, maybe have a look at `CalciteQueryTest.testFilterOnTimeExtract`?

Reply:
Thanks @clintropolis for the pointer. Moved this test to CalciteQueryTest. Added testFilterOnTimeExtractWithMilliseconds() in CalciteQueryTest. If the code changes look fine I'll go ahead and handle other time units. Introduced new data source DATASOURCE4 in tests so as to test other time units as well.

Reply:
Overall seems reasonable to me :+1:. You could probably test most/all of the additions in a single test if the timestamps for the datasource you added are sufficient, if you'd like to avoid repeating a lot of the near same boilerplate in the calcite tests.

Reply:
👍makes sense.